### PR TITLE
4.0.8.3

### PR DESCRIPTION
**Pull request by sandtechnology**

### CODE REVIEW

**Comment by sandtechnology** [comment]
Reviewed with no issue, wating @Ghost-chu

File: src/main/java/org/maxgamer/quickshop/command/subcommand/SubCommand_Reload.java [hunk 1]
```diff
@@ -35,8 +44,23 @@
...
             @NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] cmdArg) {
         MsgUtil.sendMessage(sender, MsgUtil.getMessage("command.reloading", sender));
-        plugin.getServer().getPluginManager().disablePlugin(plugin);
-        plugin.getServer().getPluginManager().enablePlugin(plugin);
+        PluginManager pluginManager = plugin.getServer().getPluginManager();
+        try {
+            File file = Paths.get(plugin.getClass().getProtectionDomain().getCodeSource().getLocation().toURI()).toFile();
+            Throwable throwable = PluginUtil.unload(plugin);
+            if (throwable != null) {
+                throw new RuntimeException("Failed to reload plugin, considering restart the server. (unload plugin failed)");
```
Comment (Ghost-chu) [inline src/main/java/org/maxgamer/quickshop/command/subcommand/SubCommand_Reload.java:52]: uppercase

File: src/main/java/org/maxgamer/quickshop/nonquickshopstuff/com/sk89q/worldedit/util/paste/EngineHubPaster.java [hunk 2]
```diff
@@ -1,78 +0,0 @@
-/*
```
Comment (Ghost-chu) [inline src/main/java/org/maxgamer/quickshop/nonquickshopstuff/com/sk89q/worldedit/util/paste/EngineHubPaster.java:1]: Should keep this file in project so we can reuse those sources code in future if we decide host same paste program by our self.

**Review by Ghost-chu — CHANGES_REQUESTED** [review]

File: src/main/java/org/maxgamer/quickshop/command/subcommand/SubCommand_Reload.java (hunk 1 above)
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/command/subcommand/SubCommand_Reload.java:52]: unload->Unload

File: src/main/java/org/maxgamer/quickshop/nonquickshopstuff/com/sk89q/worldedit/util/paste/EngineHubPaster.java (hunk 2 above)
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/nonquickshopstuff/com/sk89q/worldedit/util/paste/EngineHubPaster.java:1]: just comment whole file

**Comment by Ghost-chu** [comment]
some shit about setSignText are planned to edit in this release but not in there.

File: pom.xml [hunk 3]
```diff
@@ -836,6 +836,12 @@
             <artifactId>commons-codec</artifactId>
             <version>1.15</version>
         </dependency>
+        <dependency>
```
Comment (Ghost-chu) [inline pom.xml:839]: I don't remember I actually use this. If it used in Factions, it should provide by it.

**Review by Ghost-chu — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (sandtechnology) [reply pom.xml:839]: It's was used for compressing logs.

File: src/main/java/org/maxgamer/quickshop/util/envcheck/EnvironmentChecker.java [hunk 4]
```diff
@@ -107,12 +112,21 @@ public ResultReport run(EnvCheckEntry.Stage stage) {
                     Util.debugLog("Skip test: " + envCheckEntry.name() + ": Except stage: " + Arrays.toString(envCheckEntry.stage()) + " Current stage: " + stage);
                     continue;
                 }
-                ResultContainer executeResult = (ResultContainer) declaredMethod.invoke(this);
-                //plugin.getLogger().info("Result: "+executeResult.getResultMessage());
-                if (executeResult.getResult().ordinal() > result.ordinal()) { //set bad result if its worse than the latest one.
-                    result = executeResult.getResult();
+
+                if (!startupArgs.contains("-Dorg.maxgamer.quickshop.util.envcheck.skip." + envCheckEntry.name())) {
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/util/envcheck/EnvironmentChecker.java:116]: We should check the system properties, not the args.

File: src/main/java/org/maxgamer/quickshop/util/envcheck/EnvironmentChecker.java [hunk 5]
```diff
@@ -48,10 +49,12 @@
 public class EnvironmentChecker {
     private final QuickShop plugin;
     private final List<Method> tests = new ArrayList<>();
+    private final List<String> startupArgs;
 
     public EnvironmentChecker(QuickShop plugin) {
         this.plugin = plugin;
         this.registerTests(this.getClass()); //register self
+        this.startupArgs = ManagementFactory.getRuntimeMXBean().getInputArguments();
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/util/envcheck/EnvironmentChecker.java:57]: We should check the system properties, not the args.

Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/util/envcheck/EnvironmentChecker.java:57]: No I just want some args not all, it is useless so only check input arguments

File: src/main/java/org/maxgamer/quickshop/util/envcheck/EnvironmentChecker.java (hunk 4 above)
Comment (Ghost-chu) [reply src/main/java/org/maxgamer/quickshop/util/envcheck/EnvironmentChecker.java:116]: No I just want some args not all, it is useless so only check input arguments

**Review by Ghost-chu — APPROVED** [review]

**Pull request merged**